Angular/Angular guidelines and best practices/Assets caching.md

```diff
@@ -0,0 +1,99 @@
+> There are only two hard things in Computer Science: **cache invalidation** and naming things.
+>
+> — PHIL KARLTON
+
+After the application is built and deployed, there comes a question of caching all the various files that are required by the app.
+
+Application build files in `dist` directory include:
+
+- fingerprinted JS and CSS chunks
+- copied assets directory
+  - individual assets files are _NOT_ fingerprinted
+- index.html file
+
+## Index file
+
+The entry-point into a Single Page Application (SPA) like Angular is the `index.html` file. During the build process, the source `src/index.html` is taken and updated with links to the generated JS and CSS chunks and placed into `dist/[project-name]/index.html`.
+
+The fingerprinted links to JS and CSS chunks will be different for each build and the `index.html` file will be different as well. Because of this, it is important that the `index.html` file is not cached permanently on the client and that the client requests cache revalidation. If the client has a stale `index.html` file, he will try loading stale JS and CSS chunks as well. Those might or might not be in his cache and depending on this the client might get a completely outdated app or a broken app that is missing some chunks.
```

Comment:
```suggestion
Fingerprinted links to JS and CSS chunks will be different for each build, and the `index.html` file will be different as well. Because of this, it is important that the `index.html` file is not cached permanently on the client and that the client requests cache revalidation. If the client has a stale `index.html` file, it will try loading stale JS and CSS chunks as well. Those might or might not be in its cache, and depending on this the client might get a completely outdated app or a broken app that is missing some chunks.
```